tests/run.sh

```diff
@@ -155,19 +154,16 @@ if ! python3 -m pip show pytest-html >/dev/null; then
     echo "WARNING: install pytest-html for html results report"
 fi
 
-if [[ -n $SPECIFIC_INTEGRATION_TEST ]]; then
-    SPECIFIC_INTEGRATION_TEST_FLAG="-k"
-fi
-
 if [ -n "$K8S" ]; then
     export KUBECONFIG="${HOME}/kubeconfig.${K8S}"
     aws eks update-kubeconfig --region $AWS_DEFAULT_REGION --name $AWS_EKS_CLUSTER_NAME --kubeconfig ${HOME}/kubeconfig.${K8S}
 fi
-
+if test ${CI_NODE_TOTAL:-1} -gt 1; then
+  export PYTEST_ADDOPTS="$PYTEST_ADDOPTS -k '$(python .gitlab-ci-parallel.py $@)'"
```

Comment:
I would remove the `.` from the filename, and maybe make it pytest explicit, for example `ci-parallel-pytest-pugin.py`